File: Front.QuizApp/src/app/business/mappers/question.mapper.ts
import {Injectable} from '@angular/core';
import {QuestionDto} from '../dtos/question.dto';
import {Question} from '../models/question.model';
import {AnswerDto} from '../dtos/answer.dto';

@Injectable({providedIn: 'root'})
export class QuestionMapper {
    public mapQuestionFromApiToModel(apiQuestion: QuestionDto): Question {
        return {
            id: apiQuestion.id,
            quizId: apiQuestion.quizId,
            questionContent: apiQuestion.questionContent,
            options: apiQuestion.options,
            correctOptionIndices: []
        };
    }
}

@Injectable({providedIn: 'root'})
export class AnswerMapper {
    public mapAnswerFromApiToModel(originalQuestion: Question, apiAnswer: AnswerDto): Question {
        originalQuestion.correctOptionIndices = apiAnswer.correctOptionIndices
        return originalQuestion
    }
}

File: Back.QuizApp/Controllers/Mappers/QuestionMapper.cs
using Ynov.QuizApp.Business.Models;
using Ynov.QuizApp.Controllers.DTOs;

namespace Ynov.QuizApp.Controllers.Mappers;

public class QuestionMapper {
    public QuestionDTO ToDto(Question question) {
        return new QuestionDTO() {
            Id = question.Id,
            QuestionContent = question.QuestionContent,
            Options = question.Options
        }; 
    }
}